src/utils/PrecompiledHeaders/PrecompiledHeaders.java

```diff
@@ -0,0 +1,134 @@
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public final class PrecompiledHeaders {
+
+    private static final Pattern INCLUDE_PATTERN = Pattern.compile("^#\\s*include \"([^\"]+)\"$");
+    private static final String HOTSPOT_PATH = "src/hotspot";
+    private static final String PRECOMPILED_HPP = "src/hotspot/share/precompiled/precompiled.hpp";
+    private static final String INLINE_HPP_SUFFIX = ".inline.hpp";
+
+    private PrecompiledHeaders() {
+        throw new UnsupportedOperationException("Instances not allowed");
+    }
+
+    public static void main(String[] args) throws IOException {
+        if (args.length == 0 || args.length > 2) {
+            System.err.println("Usage: min_inclusion_count [jdk_root=.]");
+            System.exit(1);
+        }
+
+        int minInclusionCount = Integer.parseInt(args[0]);
+        Path jdkRoot = Path.of(args.length == 2 ? args[1] : ".").toAbsolutePath();
+        if (!Files.isDirectory(jdkRoot)) {
+            throw new IllegalArgumentException("jdk_root is not a directory: " + jdkRoot);
+        }
+        Path hotspotPath = jdkRoot.resolve(HOTSPOT_PATH);
+        if (!Files.isDirectory(hotspotPath)) {
+            throw new IllegalArgumentException("Invalid hotspot directory: " + hotspotPath);
+        }
+
+        // Count inclusion times for each header
+        Map<String, Integer> occurrences = new HashMap<>();
+        try (Stream<Path> paths = Files.walk(hotspotPath)) {
+            paths.filter(Files::isRegularFile)
+                    .filter(path -> {
+                        String name = path.getFileName().toString();
```

Comment:
IIUC this means that we count the number of cpp or hpp files that directly include a file, but should we count the number of cpp file that directly or transitively include a file instead?